### PR TITLE
Add content trust root-related tests

### PR DESCRIPTION
**Pull request by pkmooreanaconda**
### Description

The purpose of this PR is to add functional tests around Conda's content trust code.

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: tests/trust/test_signature_verification.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    tmp_rootdir = TemporaryDirectory()
+    with patch('conda.base.context.Context.av_data_dir', new_callable=PropertyMock) as av_data_dir_mock:
+        av_data_dir_mock.return_value = tmp_rootdir.name
+        with patch('conda.trust.signature_verification.INITIAL_TRUST_ROOT', new=_get_test_initial_trust_root()):
+            # Find 2.root.json in our test data directory...
+            testdata_2_root = path.join(_test_data_dir(), '2.root.json')
+
+            # Load 2.root.json's data so we can use it in our mock
+            with open(testdata_2_root, 'r') as f:
+                test_2_root_data = json.load(f)
```
Comment (sven6002) [inline tests/trust/test_signature_verification.py:178]: what is the difference between `2.root.json` and `3.root.json`

**Review by sven6002 — APPROVED** [review]
Looks good to me

**Review by sven6002 — APPROVED** [review]
Looks good to me..

**Comment by pkmooreanaconda** [comment]
pre-commit.ci autofix

File: conda/trust/signature_verification.py [hunk 2]
```diff
@@ -8,23 +8,24 @@
...
-        write_metadata_to_file,
-    )
-    from conda_content_trust.signing import wrap_as_signable
-except ImportError:
-    # SignatureStatus.enabled handles this
-    pass
+#    from conda_content_trust.signing import wrap_as_signable
+# except ImportError:
+#    # SignatureStatus.enabled handles this
+#    pass
```
Comment (kenodegard) [inline conda/trust/signature_verification.py:28]: This is causing the flake8 error

**Review by kenodegard — APPROVED** [review]

**Comment by jezdez** [comment]
Wondering if https://github.com/conda/conda/issues/11805 is (partially) solved by that?

**Pull request merged**